### PR TITLE
Test vm creation

### PR DESCRIPTION
**Pull request by kamilash**

### CODE REVIEW

File: tests/lib/azure/vm.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		}
+		osDisk := *properties.StorageProfile.OsDisk
+		gotOsType := string(osDisk.OsType)
+		if gotOsType != expectedOsType {
+			return errors.New("OS type expected is " + expectedOsType + " but got " + gotOsType)
+		}
+		if properties.NetworkProfile == nil {
+			return errors.New("Field NetworkProfile is nil!")
+		}
+		network := *properties.NetworkProfile.NetworkInterfaces
```
Comment (matheusvill) [inline tests/lib/azure/vm.go]: You don't check if this resource isn't nil.

File: tests/azure/testdata/key.pub [hunk 2]
```diff
@@ -0,0 +1 @@
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC5CsM1AZ/4xXPC2h9EKrOQWtgAkWRAQO0j4kHnPyRuCio/+5GVjc4l0sFUPHLMihAzmtQSWl/oRAEeqGXpDyzvuDhQb9+xuci6egoEY/uEEGHsD12R3/VdtYECmjLNS/JnopT5BptvzBuPXlomYMv3yL3O3dV+v/F6WNqtZ2LjOYAXLWjw1BrW6tA2/JIh2H0g7Dssa739wewm38AKRCGOpKSLnZN8KK43uwAl69oNa2BUJgqdWfzdTIQFOetxZf8Fe+4UB0lwjE8DfPfYQHmMJA/sy2GRoDoR+2pWVVO2SQx/+AxjcxPjICuF8Toac6ImzV9dcaaKyFQr2NOKJd6x kamila@kamila-Latitude-3450
```
Comment (katcipis) [inline tests/azure/testdata/key.pub]: kamila@kamila-Latitude-3450 :-P

File: tests/lib/azure/vm.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		if len(network) == 0 {
+			return errors.New("Field NetworkInterfaces is nil!")
+		}
+		net := network[0]
+		gotNic := string(*net.ID)
```
Comment (katcipis) [inline tests/lib/azure/vm.go]: No chance of a nil pointer explosion here ?

Comment (katcipis) [inline tests/lib/azure/vm.go]: All this validation seems annoying as hell...I think one change on Go 1.8 may help us, structs with different tags are now equal, so we could re-declare our version of the struct with tags for struct validation and use some sort of struct validation for this. The downside is that we will require Go 1.8 to run the tests...which I'm ok with :-)

Comment (katcipis) [inline tests/lib/azure/vm.go]: fatih has a projetct that perhaps would help too: https://github.com/fatih/structs
on this case it seems that checking fields on a map would involve a lot less if'ing around

File: tests/lib/azure/vm.go (hunk 1 above)
Comment (katcipis) [inline tests/lib/azure/vm.go]: if NetworkInterfaces is a slice it is not necessary to check for nil, just check its len ;-). Nil slices are valid empty slices.

File: tests/azure/vm_test.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+		osDisk,
+		imageUrn,
+		customData,
+		keyFile,
+	)
+	vms := azure.NewVM(f)
+	vms.AssertExists(t, vm, availSet, vmSize, osType, nic)
+}
+
+func createVMResources(f fixture.F) (string, string, string, string, string) {
```
Comment (tiago4orion) [inline tests/azure/vm_test.go]: or maybe a good use case for `named returns`.

File: tests/lib/azure/vm.go (hunk 3 above)
Comment (tiago4orion) [inline tests/lib/azure/vm.go]: I see `reflect` ... everywhere...

Comment (tiago4orion) [inline tests/lib/azure/vm.go]: Azure API sucks .... Why pointers in the slice entries?

File: tests/azure/vm_test.go (hunk 4 above)
Comment (kamilash) [inline tests/azure/vm_test.go]: But would still have 5 parameters to return. Do you think named returns would be better in this case?
Well, already created a struct, but can change again for named returns if it is better.

Comment (tiago4orion) [inline tests/azure/vm_test.go]: I don't think so. A struct is better =)

**Pull request merged**